src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,855 +1,245 @@
+  </TabItem>
+  <TabItem label="context.params">
+    ```ts title="src/pages/posts/[id].json.ts" "params:" {11-15}
+    import type { APIContext } from 'astro';
+
+    export function getStaticPaths() {
+      return [
+        { params: { id: '1' } },
+        { params: { id: '2' } },
+        { params: { id: '3' } }
+      ];
+    }
+
+    export function GET({ params }: APIContext) {
+      return new Response(
+        JSON.stringify({ id: params.id }),
+      );
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+
+When pages are rendered on demand, `params` can be any value matching the path segments in the dynamic route pattern.
+
+```astro title="src/pages/posts/[id].astro" "Astro.params"
 ---
-<Fragment set:html={html} />
-```
+import { getPost } from '../api';
 
-A callback function passed as `<Shout />`’s child will receive the all-caps `message` parameter:
+const post = await getPost(Astro.params.id);
 
-```astro title="src/pages/index.astro"
----
-import Shout from "../components/Shout.astro";
+// No posts found with this ID
+if (!post) {
+  return Astro.redirect("/404")
+}
 ---
-<Shout message="slots!">
-  {(message) => <div>{message}</div>}
-</Shout>
-
-<!-- renders as <div>SLOTS!</div> -->
-```
-
-Callback functions can be passed to named slots inside a wrapping HTML element tag with a `slot` attribute. This element is only used to transfer the callback to a named slot and will not be rendered onto the page.
-
-```astro
-<Shout message="slots!">
-  <fragment slot="message">
-    {(message) => <div>{message}</div>}
-  </fragment>
-</Shout>
+<html>
+  <h1>{post.name}</h1>
+</html>
 ```
 
-Use a standard HTML element for the wrapping tag, or any lower case tag (e.g. `<fragment>` instead of `<Fragment />`) that will not be interpreted as a component.  Do not use the HTML `<slot>` element as this will be interpreted as an Astro slot.
+See also: [`params`](/en/reference/routing-reference/#params)
 
-### `Astro.self`
+### `url`
 
-`Astro.self` allows Astro components to be recursively called. This behaviour lets you render an Astro component from within itself by using `<Astro.self>` in the component template. This can be helpful for iterating over large data stores and nested data-structures.
+<p>
 
-```astro
----
-// NestedList.astro
-const { items } = Astro.props;
----
-<ul class="nested-list">
-  {items.map((item) => (
-    <li>
-      <!-- If there is a nested data-structure we render `<Astro.self>` -->
-      <!-- and can pass props through with the recursive call -->
-      {Array.isArray(item) ? (
-        <Astro.self items={item} />
-      ) : (
-        item
-      )}
-    </li>
-  ))}
-</ul>
-```
+**Type:** `URL`<br />
+<Since v="1.0.0" />
+</p>
 
-This component could then be used like this:
+`url` is a [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object constructed from the current `request.url` URL string value. It is useful for interacting with individual properties of the request URL, like pathname and origin. 
```

Comment:
I wonder if the sentence is a bit "heavy" (hard to read) because of the repetition of `URL` (lower/upper). Would something like this work:
```suggestion
`url` is a [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object constructed from the current `request.url` value. It is useful for interacting with individual properties of the request URL, like pathname and origin. 
```
That only takes away one... and I understand that this was to emphasize that the value is a string, but is it useful here? I don't think we need to know that the value used to compute `Astro.url` is a string... So maybe this help to lighten the sentence.